### PR TITLE
Migrate from travis to github actions

### PR DESCRIPTION
**Pull request by bilalawan321**
[LEARNER-8451](https://openedx.atlassian.net/browse/LEARNER-8451l)

As Github Actions is enabled in the test repo [https://github.com/edx/edx-app-test/pull/86](https://github.com/edx/edx-app-test/pull/86)

In this PR, I will remove all Travis files and add Travis code to Github actions Job and make sure that GitHub actions job is triggered on every push and pull request.

### CODE REVIEW

File: .github/workflows/github_actions_demo.yml [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
     runs-on: ubuntu-latest
     steps:
-      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
-      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by GitHub!"
-      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
-      - run: echo "🍏 This job's status is ${{ job.status }}."
+    - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
+    - uses: actions/checkout@v2
+    - name: Install Dependencies
+      run: |
```
Comment (mraarif) [inline .github/workflows/github_actions_demo.yml:12]: this should be done in separate steps, and we should ideally be installing the requirements from a requirements file

**Review by mraarif — COMMENTED** [review]
few changes that I'd recommend,
1. should rename the workflow file to be something like `ci.yml`
2. should set up a matrix to tests against multiple python versions as done in Travis CI setup, docs on `matrix` here: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix
3. `pip-tools` would require an `.in` file to generate a requirement `.txt` file from, we should instead rename `travis.in` to `ci.in` instead of deleting it, and update the `upgrade` target in `Makefile` to reflect the change

File: .github/workflows/github_actions_demo.yml (hunk 1 above)
Comment (bilalawan321) [reply .github/workflows/github_actions_demo.yml]: thank you, I have resolved all your given points, can you plz take a look. ?

**Review by mraarif — APPROVED** [review]
please squash the commits before you merge

**Pull request merged**